packages/netlify-cms-core/src/components/Editor/EditorControlPane/EditorControl.js

```diff
@@ -115,13 +115,31 @@ class EditorControl extends React.Component {
     t: PropTypes.func.isRequired,
     isEditorComponent: PropTypes.bool,
     isNewEditorComponent: PropTypes.bool,
+    parentIds: PropTypes.arrayOf(PropTypes.string),
+  };
+
+  static defaultProps = {
+    parentIds: [],
   };
 
   state = {
     activeLabel: false,
   };
 
   uniqueFieldId = uniqueId(`${this.props.field.get('name')}-field-`);
+  isList = this.props.field.get('widget') === 'list';
+  isListOrObject = this.isList || this.props.field.get('widget') === 'object';
+
+  isAncestorOfFieldError = () => {
+    const { fieldsErrors } = this.props;
+
+    if (this.isListOrObject && fieldsErrors.size > 0) {
```

Comment:
We should avoid hardcoding widget names in the core. Is `isListOrObject` required?

Reply:
I just put it there as a potential optimization so that it didn't need to check the parentIds for every single field, but it's probably not necessary.